### ISSUE TITLE
Data Generator Module

### ISSUE THREAD

**Issue by dhblum** [ibpsa/project1-boptest#55]
This issue addresses the development of a data generator, which is used to coordinate the use of test case data for the Forecast and KPI Calculation modules, as well as export of an FMU from a test case directory.

To Do:
- [x] Getting variable data for Forecast and KPI Calculation from multiple CSVs in the FMU
- [x] Explore addition to parser to integrate FMU export from test case.
- [x] Finish definition of variable names and add check that KPI json output variables have corresponding values in data csv (for instance for Forecaster to retrieve)
- [ ] Create 2nd year weather data for MPC model identification

**Comment by JavierArroyoBastida** [comment]
@dhblum, a few notes from our discussions this week to keep them in the history. @mwetter could you please review them?

- The `generate_data` method of the data generator should be called from the `export_fmu` method of the parser since this method needs to be executed only once in the creation of the test case. 
- The `generate_data` method will inspect the variables of all files in the testcase#/models/Resources folder to look for test case data. Every file name is possible but a convention is stablished for the variables to be integrated in the test case data and all files should contain a time column. 
- The data will be stored within the resources folder of the wrapped.fmu in different CSVs with different sampling time periods.
- The sampling period to be used is the one provided by the original data file in the testcase#/models/Resources folder. 
- A forward fill resampling will be applied for all variables except for the weather data.
- The weather data will be provided in a sampling of one minute since it requires spline interpolation that is computed in the weatherTMY3 reader. @dhblum and I concluded that re-implementing the spline interpolation in the forecaster may be an unnecessary code duplication.
- The generation of data for system identification should allow the user to use his/her own control input signals but under different boundary conditions than the standard boundary conditions of the test case. For that the following is required:
  1. The generation of a second weather file from the first one. To be done with the code proposed by Krzysztof.
  2. Enabling wrapped.fmu to dynamically pick the desired weather file (either weather standard or weather system id) when calling `advance`. This has proven particularly challenging because the absolute path is defined in the model compilation. For the data generator we hard-code a change in the filNam parameter of the ReaderTMY3 model to point to the file before compiling ReaderTMY3.fmu . However, here we cannot do the same because wrapped.fmu is already compiled. Any idea how to solve this @mwetter?

**Comment by dhblum** [comment]
2nd year weather data for MPC model identification will be addressed in its own issue.

**Comment by dhblum** [comment]
Closed by https://github.com/ibpsa/project1-boptest/pull/61.
